src/service/angular-sails.js

```diff
@@ -1,74 +1,238 @@
-    var provider = this,
-        httpVerbs = ['get', 'post', 'put', 'delete'],
-        eventNames = ['on', 'once'];
+(function(angular, io) {
+  'use strict';
+  io.sails.autoConnect = false;
 
+  // copied from angular
+  function parseHeaders(headers) {
+    var parsed = Object.create(null), key, val, i;
+    if(!headers) return parsed;
+    forEach(headers.split('\n'), function(line) {
+      i = line.indexOf(':');
+      key = lowercase(trim(line.substr(0, i)));
+      val = trim(line.substr(i + 1));
+      if(key) {
+        parsed[key] = parsed[key] ? parsed[key] + ', ' + val : val;
+      }
+    });
+
+    return parsed;
+  }
+  function trim(value) {
+    return angular.isString(value) ? value.trim() : value;
+  }
+
+  // copied from angular
+  function headersGetter(headers) {
+    var headersObj = angular.isObject(headers) ? headers : undefined;
+    return function(name) {
+      if(!headersObj) headersObj = parseHeaders(headers);
+      if(name) {
+        var value = headersObj[lowercase(name)];
+        if(value === void 0) {
+          value = null;
+        }
+        return value;
+      }
+      return headersObj;
+    };
+  }
+
+
+  /*global angular */
+  angular.module('ngSails', ['ng']);
+
+  /*global angular, io */
+  angular.module('ngSails').provider('$sails', function() {
+    var provider = this;
+
+    // wrap these socket.io methods with angular promises
+    this.httpVerbs = ['get', 'post', 'put', 'delete'];
+
+    // wrap these events in $evalAsync to fire a digest cycle
+    this.eventNames = ['on', 'once'];
+
+    // the url to connect to
     this.url = undefined;
-    this.interceptors = [];
-    this.responseHandler = undefined;
-
-    this.$get = ['$q', '$timeout', function ($q, $timeout) {
-        var socket = io.connect(provider.url),
-            defer = function () {
-                var deferred = $q.defer(),
-                    promise = deferred.promise;
-
-                promise.success = function (fn) {
-                    promise.then(fn);
-                    return promise;
-                };
-
-                promise.error = function (fn) {
-                    promise.then(null, fn);
-                    return promise;
-                };
-
-                return deferred;
-            },
-            resolveOrReject = this.responseHandler || function (deferred, data) {
-                // Make sure what is passed is an object that has a status that is a number and if that status is no 2xx, reject.
-                if (data && angular.isObject(data) && data.status && !isNaN(data.status) && Math.floor(data.status / 100) !== 2) {
-                    deferred.reject(data);
-                } else {
-                    deferred.resolve(data);
+    
+    // Prefix every request with this url
+    this.urlPrefix = '';
+    
+    this.config = {
+      // Transports to use when communicating with the server, in the order they will be tried
+      transports: ['websocket', 'polling'],
```

Comment:
dont know, `['polling', 'websocket']` is the io.sails standard, i only changed the order